packages/web-components/src/switch/switch.spec.ts

```diff
@@ -0,0 +1,461 @@
+        `;
+    });
+
+    const hasAriaReadonlyAttribute = await element.evaluate((node: Element) => node.hasAttribute('aria-readonly'));
+    expect(hasAriaReadonlyAttribute).toBe(false);
+  });
+
+  test('should set the `aria-checked` attribute equal to the `checked` property', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-switch></fluent-switch>
+        `;
+    });
+
+    await element.evaluate((node: Switch) => {
+      node.checked = true;
+    });
+
+    await expect(element).toHaveAttribute('aria-checked', 'true');
+
+    await element.evaluate((node: Switch) => {
+      node.checked = false;
+    });
+
+    await expect(element).toHaveAttribute('aria-checked', 'false');
+  });
+
+  test('should set the `aria-readonly` attribute equal to the `readonly` value', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-switch></fluent-switch>
+        `;
+    });
+
+    await element.evaluate((node: Switch) => {
+      node.readOnly = true;
+    });
+
+    await expect(element).toHaveAttribute('aria-readonly', 'true');
+
+    await element.evaluate((node: Switch) => {
+      node.readOnly = false;
+    });
+
+    await expect(element).toHaveAttribute('aria-readonly', 'false');
+  });
+
+  test('should initialize to the initial value if no value property is set', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-switch></fluent-switch>
+        `;
+    });
+
+    const initialValue = await element.evaluate<string, Switch>(node => node.initialValue);
+
+    await expect(element).toHaveJSProperty('value', initialValue);
+  });
+
+  test('should add a class of `label` to the internal label when default slotted content exists', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-switch></fluent-switch>
+        `;
+    });
+    const label = element.locator('.label');
+
+    await element.evaluate(node => {
+      node.innerHTML = 'Label';
+    });
+
+    await expect(label).toHaveClass(/label/);
+
+    await expect(label).not.toHaveClass(/label__hidden/);
+  });
+
+  test('should add classes of `label` and `label__hidden` to the internal label when default slotted content exists', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-switch>Switch</fluent-switch>
+        `;
+    });
+
+    const label = element.locator('.label');
+
+    await element.evaluate(node => {
+      node.innerHTML = '';
+    });
+
+    await expect(label).toHaveClass(/label/);
+
+    await expect(label).toHaveClass(/label__hidden/);
+  });
+
+  test('should set the `aria-disabled` attribute equal to the `disabled` value', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-switch></fluent-switch>
+        `;
+    });
+
```

Comment:
I would suggest to also test the value of the attribute (in this case `aria-disabled`) after the initial default render. Similar to what is implemented in the next test (should NOT set a tabindex when disabled is `true`).